source/common/config/protobuf_link_hacks.h

```diff
@@ -43,5 +45,19 @@ const envoy::service::listener::v3alpha::LdsDummy _lds_dummy_v3;
 const envoy::service::route::v3alpha::RdsDummy _rds_dummy_v3;
 const envoy::service::cluster::v3alpha::CdsDummy _cds_dummy_v3;
 const envoy::service::endpoint::v3alpha::EdsDummy _eds_dummy_v3;
-const envoy::service::route::v3alpha::SrdsDummy _srds_dummy_v4;
+const envoy::service::route::v3alpha::SrdsDummy _srds_dummy_v3;
+
+// With the v2 -> v3 migration there is another, related linking issue.
+// Symbols for v2 protos which headers are not included in any file in the codebase are being
+// dropped by the linker in some circumstances. For example, in the Envoy Mobile iOS build system.
+// Even though all v2 packages are included as a dependency in their corresponding v3 package, and
+// `always_link` is set for all proto bazel targets.
+// Further proof of this can be seen by way of counter example with the envoy.api.v2.Cluster type,
+// which is checked for by proto_descriptors.cc. This type **is** getting linked because its headers
+// is still included in cds_api_impl.cc. On the other side because the v2 hds header is not included
+// anywhere the v2 service type is getting dropped, and thus the descriptor is not present in the
+// descriptor pool.
```

Comment:
Can you include a link to the tracking issue here?

Reply:
yep, was waiting to see what you thought. Will create and link now.